source/mir/math/stat.d

```diff
@@ -420,6 +424,202 @@ unittest
     assert(float.max.repeat(3).hmean!(double, "fast").approxEqual(float.max));
 }
 
+private template centerImpl(F)
+{
+    import mir.ndslice.slice: Slice, SliceKind;
+
+    @safe pure nothrow @nogc
+    auto centerImpl(Iterator, size_t N, SliceKind kind)(
+        Slice!(Iterator, N, kind) slice, F average)
+    {
+        return slice - average;
+    }
+}
+
+/++
+Centers `r`, which must be a finite iterable.
+
+By default, `r` is centered by the mean. A custom function may also be provided
+using `centralTendency`.
+
+Returns:
+    The elements in the range r with the average subtracted from them.
++/
+template center(F, alias centralTendency = mean!(F, Summation.appropriate))
+{
+    /++
+    Params:
+        r = range
+    +/
+    @safe pure nothrow
+    auto center(Range)(Range r)
+        if (isIterable!Range)
+    {
+        return centerImpl!F(r, centralTendency(r));
+    }
+    
+    /++
+    Params:
+        r = range
+        average = average
+    +/
+    @safe pure nothrow
+    auto center(Range)(Range r, out F average)
```

Comment:
let's remove this overload.
The reason is that the user needs to use the two lines anyway.

```d
F a;
auto centered = r.center(a);
```

vs

```d
auto a = r.mean;
auto centered = r - a;
```

But we can preserve the examples.

Also, we don't really need two outer templates. `centered!(mean!(double, "precise"))` should just work.